### PR TITLE
Document graphql

### PR DESCRIPTION
**Pull request by ayyazdaniaryan**
This shows the new documentation of our GraphQL APIs. They're not perfect, but they were a lot of APIs to report and document on. From now on, any graphQL endpoint that is created should have documentation set something like this. We've tried to collate everything in one place. Open to changes, and improvements as the come.

### CODE REVIEW

File: assembl/graphql/docstrings.py [hunk 1]
```diff
@@ -0,0 +1,961 @@
...
+    discussion = """The discussion object metadata."""
+    landing_page_module_types = """The metadata object for LandingPageModule object."""
+    landing_page_modules = """A list of LandingPageModules."""
+    text_fields = """A list of ConfigurableField union, where each text field represents a field on a bound entity."""
+    profile_fields = """A list of ConfigurableField union, where each text field represents a field on a profile only."""
+    timeline = """A list of DiscussionPhase objects, descriping the timeline objects on the debate."""
+
+
+class Discussion:
+    __docs__ = """The Discussion object. This object describes certain parts of the core state of the debate."""
```
Comment (vincentfretin) [inline assembl/graphql/docstrings.py]: you wrote `__docs__` instead of `__doc__` on purpose?

File: assembl/graphql/docstrings.py [hunk 2]
```diff
@@ -0,0 +1,961 @@
...
+    homepage_url = """A URL for the homepage (optional). Often placed on the logo."""
+    title = """The title of the discussion, in the language specified by the input"""
+    subtitle = """The subtitle of the discussion, in the language specified by the input"""
+    button_label = """The value inside of the participation button in the landing page."""
+    header_image = Default.document % ("The file representing the header of the landing page. ", )
+    logo_image = Default.document % ("The file representing the logo of the debate. ", )
+
+
+class UpdateDiscussion:
+    __doc__ = """The mutation that allows to update an existing Discussion object"""
```
Comment (vincentfretin) [inline assembl/graphql/docstrings.py]: You can just write the docstring on the class, without `__doc__ = `. The string will be available on the `__doc__` class attribute.

```
>>> class A(object):
...   """lalala"""
... 
>>> A.__doc__
'lalala'
```

File: assembl/graphql/landing_page.py [hunk 3]
```diff
@@ -32,18 +34,24 @@ def resolve_title_entries(self, args, context, info):
...
     module_type = graphene.Field(LandingPageModuleType)
     exists_in_database = graphene.Boolean()
     title = graphene.String(lang=graphene.String())
     title_entries = graphene.List(LangStringEntry)
     subtitle = graphene.String(lang=graphene.String())
     subtitle_entries = graphene.List(LangStringEntry)
+=======
+    module_type = graphene.Field(LandingPageModuleType, description=docs.LandingPageModule.module_type)
+    exists_in_database = graphene.Boolean(description=docs.LandingPageModule.exists_in_database)
+>>>>>>> Update landing page docstrings
```
Comment (vincentfretin) [inline assembl/graphql/landing_page.py]: You seems to have left a conflict here.

File: assembl/graphql/post.py [hunk 4]
```diff
@@ -73,39 +78,41 @@ def resolve_creator(self, args, context, info):
...
     def resolve_extracts(self, args, context, info):
         return self.db.query(models.Extract
-            ).join(models.Content, models.Extract.content == self
-            ).options(joinedload(models.Extract.text_fragment_identifiers)
-            ).order_by(models.Extract.creation_date
-            ).all()
+                             ).join(models.Content, models.Extract.content == self
+                                    ).options(joinedload(models.Extract.text_fragment_identifiers)
+                                              ).order_by(models.Extract.creation_date
+                                                         ).all()
```
Comment (vincentfretin) [inline assembl/graphql/post.py]: I don't like this change.

File: assembl/graphql/landing_page.py (hunk 3 above)
Comment (ayyazdaniaryan) [reply assembl/graphql/landing_page.py]: Oops, that must have been left from the rebase. Fixing now.

File: assembl/graphql/post.py (hunk 4 above)
Comment (ayyazdaniaryan) [reply assembl/graphql/post.py]: It's an auto-pep8 service that does this on another dev's machine.

File: assembl/graphql/docstrings.py (hunk 1 above)
Comment (ayyazdaniaryan) [reply assembl/graphql/docstrings.py]: Nope. error, good catch.

File: assembl/graphql/post.py (hunk 4 above)
Comment (vincentfretin) [reply assembl/graphql/post.py]: You should configure a pep8 config to not allow this indentation. There is no issue with flake8.

File: assembl/graphql/docstrings.py (hunk 2 above)
Comment (ayyazdaniaryan) [reply assembl/graphql/docstrings.py]: Yes, absolutely. The reason for this was to keep the editable string of text for document in one place as we one-shot this. Also, for future, any non-dev or front-end devs can edit documentation without python knowledge.

**Pull request merged**